### ISSUE TITLE
Add TypeScript typings

### ISSUE THREAD

**Issue by SimenB** [guybedford/cjs-module-lexer#15]
It would be very nice if this module came with TS typings 🙂

**Comment by guybedford** [comment]
I've marked this as an enhancement issue and will leave it open here. If anyone would like to contribute please do.

**Comment by SimenB** [comment]
I was gonna send a PR since the types aren't too advanced, but it seems the functions are different between CJS and ESM with CJS having `parse` and ESM having `{init, parse}`.

Thoughts on changing the API to be the same across module types? I.e. changing CJS to

```js
module.exports.parse = parseCJS;
module.exports.init = () => Promise.reject('init is not supported when using CommonJS');
```

Could potentially just no-op in the last one, or warn?

**Comment by guybedford** [comment]
@SimenB I can get behind unifying the APIs to do that sure. That sounds like a good suggestion to me.

**Comment by SimenB** [comment]
Cool. I can send a couple of PRs tomorrow if you don't get to it.

Current thinking is `init` should just be `Promise.resolve()` so you know you can always call it. Without worrying about what API

**Comment by SimenB** [comment]
Some time freed up tonight, so sent #27 & #28 👍

**Comment by guybedford** [comment]
Thanks for the PRs! I've published a 0.6.0 with these.

**Comment by SimenB** [comment]
Thanks @guybedford!
